spring-pulsar/src/main/java/org/springframework/pulsar/config/MethodPulsarListenerEndpoint.java

```diff
@@ -140,15 +144,26 @@ protected AbstractPulsarMessageToSpringMessageAdapter<V> createMessageListener(
 
 		ConcurrentPulsarMessageListenerContainer<?> containerInstance = (ConcurrentPulsarMessageListenerContainer<?>) container;
 		PulsarContainerProperties pulsarContainerProperties = containerInstance.getContainerProperties();
+
+		// Resolve the schema using the listener schema type
 		SchemaResolver schemaResolver = pulsarContainerProperties.getSchemaResolver();
 		SchemaType schemaType = pulsarContainerProperties.getSchemaType();
 		ResolvableType messageType = resolvableType(messageParameter);
-		schemaResolver.resolveSchema(schemaType, messageType).ifResolved(pulsarContainerProperties::setSchema);
-
+		schemaResolver.resolveSchema(schemaType, messageType)
+			.ifResolvedOrElse(pulsarContainerProperties::setSchema,
+					(ex) -> this.logger
+						.warn(() -> "Failed to resolve schema for type %s - will default to BYTES (due to: %s)"
+							.formatted(schemaType, ex.getMessage())));
 		// Make sure the schemaType is updated to match the current schema
 		if (pulsarContainerProperties.getSchema() != null) {
-			SchemaType type = pulsarContainerProperties.getSchema().getSchemaInfo().getType();
-			pulsarContainerProperties.setSchemaType(type);
+			// AUTO_CONSUME schema info is null - but we know it should be AUTO_CONSUME
```

Comment:
Is it necessary to set the SchemaType again explicitly? As far as I can see it's derived from `pulsarContainerProperties` which come from the `PulsarListener` annotation, which should always be AUTO_CONSUME in this case. What am I missing?

Reply:
This was originally here to deal w/ the fact that the input schema type to the resolver may result in a schema that is not necesarrilly the "same" as the type. Namely, the `SchemaType.NONE` is set on the listener which will result in a schema based on the input message type. 

However, in the special `AUTO_CONSUME` case it has no schemaInfo in it and I was just bypassing the logic but as you pointed out in this case it will be the proper SchemaType to Schema relation and I will adjust this accordingly and also add a better comment about why its here in the first place. 

Good catch! Thanks.

Reply:
Additionally this chunk of logic really needs to be factored out into a common base or delegate as it is highly duplicated amongst the MethodPulsarListener/MethodPulsarReader/ReactiveMethodPulsarListener[Endpoint] classes.

Reply:
Updated in each of the MethodReactive/PulsarListener/ReaderEndpoints.